src/M24SR.cpp

```diff
@@ -0,0 +1,1956 @@
+		(*NbByte) += Command->Body.LC;
+	}
+	/* add Le field  */
+	if ((CommandStructure & M24SR_LE_NEEDED) != 0) {
+		pCommand[(*NbByte)++] = Command->Body.LE;
+	}
+	/* add CRC field  */
+	if ((CommandStructure & M24SR_CRC_NEEDED) != 0) {
+		uCRC16 = M24SR_ComputeCrc(pCommand, (uint8_t) (*NbByte));
+		/* append the CRC16 */
+		pCommand[(*NbByte)++] = GETLSB(uCRC16);
+		pCommand[(*NbByte)++] = GETMSB(uCRC16);
+	}
+
+}
+
+
+
+
+/**
+ * @brief  	This function returns M24SR_STATUS_SUCCESS if the pBuffer is an s-block
+ * @param  	pBuffer		:	pointer to the data
+ * @retval 	M24SR_SUCCESS  :  the data is a S-Block
+ * @retval 	NFC_ERROR  	:  the data is not a S-Block
+ */
+ static M24SR::StatusTypeDef IsSBlock(uint8_t *pBuffer) {
+
+	if ((pBuffer[M24SR_OFFSET_PCB] & M24SR_MASK_BLOCK) == M24SR_MASK_SBLOCK) {
+		return M24SR::M24SR_SUCCESS;
+	} else {
+		return M24SR::M24SR_ERROR;
+	}
+
+}
+
+
+M24SR::M24SR(const uint8_t address, TwoWire *I2C,gpoEventCallback eventCallback, const int GPOPinName,
+		const int RFDISPinName) :
+		who_am_i(0),
+		type(0),
+		address(address),
+		dev_I2C(I2C),
+		GPOPin(GPOPinName),
+		RFDisablePin(RFDISPinName),
+		mCommunicationType(SYNC),
+		mLastCommandSend(NONE),
+		mCallback(&defaultCallback),
+		mComponentCallback(NULL),
+		mNDefTagUtil(new NDefNfcTagM24SR(*this)),
+		mManageGPOCallback(*this),
+		mReadIDCallback(*this){
+		//mNDefTagUtil(NULL){
+	memset(uM24SRbuffer, 0, 0xFF * sizeof(int8_t));
+	uDIDbyte = 0;
+    pinMode(RFDisablePin, OUTPUT);
+    digitalWrite(RFDisablePin, 0);
+    pinMode(GPOPin, INPUT_PULLUP);
+    if(eventCallback!=NULL) {
+      attachInterrupt(GPOPin, eventCallback, FALLING);
+    }
+}
+
+M24SR::~M24SR(){
+	delete mNDefTagUtil;
+}
+
+/**
+ * @brief  This function sends the FWT extension command (S-Block format)
+ * @param	FWTbyte : FWT value
+ * @return M24SR_SUCCESS if no errors
+ */
+ M24SR::StatusTypeDef M24SR::M24SR_SendFWTExtension(uint8_t FWTbyte) {
+	uint8_t pBuffer[M24SR_STATUSRESPONSE_NBBYTE];
+	M24SR::StatusTypeDef status;
+	uint8_t NthByte = 0;
+	uint16_t uCRC16;
+
+	/* create the response */
+	pBuffer[NthByte++] = 0xF2;
+	pBuffer[NthByte++] = FWTbyte;
+	/* compute the CRC */
+	uCRC16 = M24SR_ComputeCrc(pBuffer, 0x02);
+	/* append the CRC16 */
+	pBuffer[NthByte++] = GETLSB(uCRC16);
+	pBuffer[NthByte++] = GETMSB(uCRC16);
+
+	/* send the request */
+	status = M24SR_IO_SendI2Ccommand(NthByte, pBuffer);
+	if(status != M24SR_SUCCESS)
+		return status;
+
+	mLastCommandSend=UPDATE;
+
+	if(mCommunicationType==M24SR::SYNC){
+		return M24SR_ReceiveUpdateBinary();
+	}//if
+
+	return M24SR_SUCCESS;
+}
+
+/**
+ * @brief  This function initialize the M24SR device
+ * @retval None
+ */
+ M24SR::StatusTypeDef M24SR::M24SR_Init(M24SR_InitTypeDef *notUsed) {
+	(void) notUsed;
+	//force to open a i2c session
+	StatusTypeDef status = M24SR_ForceSession();
+	if(status!= M24SR_SUCCESS)
+		return status;
+	//leave the gpo always up
+
+    status = M24SR_ManageI2CGPO(DEFAULT_GPO_STATUS);

```

Comment:
Please review indentation, sometimes tab, sometimes 2 spaces or more.
Preferred space instead of tab.
2 spaces are usually used in the core.